chromadb/test/property/test_embeddings.py

```diff
@@ -0,0 +1,211 @@
+import pytest
+import logging
+from hypothesis import given, assume, settings, note
+import hypothesis.strategies as st
+from typing import List, Set, TypedDict, Sequence
+import chromadb
+from chromadb.api import API
+from chromadb.api.models.Collection import Collection
+from chromadb.test.configurations import configurations
+import chromadb.api.types as types
+import chromadb.test.property.strategies as strategies
+import numpy as np
+import numpy
+from hypothesis.stateful import (
+    Bundle,
+    RuleBasedStateMachine,
+    rule,
+    initialize,
+    precondition,
+    consumes,
+    run_state_machine_as_test,
+    multiple,
+    invariant,
+)
+from collections import defaultdict
+import time
+import chromadb.test.property.invariants as invariants
+
+
+traces = defaultdict(lambda: 0)
+
+
+def trace(key):
+    global traces
+    traces[key] += 1
+
+
+def print_traces():
+    global traces
+    for key, value in traces.items():
+        print(f"{key}: {value}")
+
+
+@pytest.fixture(scope="module", params=configurations())
+def api(request):
+    configuration = request.param
+    return chromadb.Client(configuration)
+
+
+dtype_st = st.shared(st.sampled_from(strategies.float_types), key="dtype")
+dimension_st = st.shared(st.integers(min_value=2, max_value=2048), key="dimension")
+
+
+class PopulatedEmbeddingSet(TypedDict):
```

Comment:
What is a "Populated embedding set" vs an embedding set?

Reply:
Where the metadatas and documents are always a list and not None, to satisfy the type checker.

Reply:
ok thanks, can we leave a quick comment explaining that. I was trying to pull meaning out of it

Reply:
Actually if I remove this and just leave the types implicit from the literal value, the type checker realizes it's ok and stops complaining.